### ISSUE TITLE
[v2 FEATURE]: Staged fine-tuning for freezing layers for from-foundation models

### ISSUE THREAD

**Issue by lewismervin1** [chemprop/chemprop#1305]
**Is your feature request related to a problem? Please describe.**
Adding customizable parameters which allow freezing layers during fine-tuning of foundation models should improve performance

Reasoning

Early in training, the randomly initialized feed-forward network (FFN) needs to adapt to the stable molecular embeddings learned by the foundation models' message-passing network (MPNN).

If you do not freeze or unfreeze too early, gradients from the untrained FFN can distort the pretrained MPNN weights.

Once the FFN converges somewhat (loss stabilizes, validation improves), unfreezing the MPNN allows small adjustments that align structural features with the new target.

Typical schedule

Phase 1: Freeze all MPNN layers. Train only the FFN for a few epochs (5–20 depending on dataset size and loss convergence).

Phase 2: Unfreeze the last one or all MPNN layers. Fine-tune jointly with a lower learning rate (e.g. 1 / 10 of the initial).

Phase 3: Optionally unfreeze all layers if domain shift is large or plateau persists.

This approach mirrors “gradual unfreezing” used in transfer learning for language and vision models and works well in Chemprop when the pretraining and fine-tuning tasks are related.

**Use-cases/examples of this new feature**
Useful for cases when training from-foundation model

**Desired solution/workflow**
Options for how many epochs to train with freezing/ without freezing the MPNN layers for Phase 1-3.

**Discussion**

**Additional context**

**Comment by JacksonBurns** [comment]
@lewismervin1 thanks for the thorough suggestions. I'll say that, in my own use of Chemprop, I've found a procedure like this to be useful.

I'll say that this is probably a bit too complicated to implement into the Chemprop CLI. Do you have citations for this fine-tuning procedure, preferably some which show performance improvement? I ask because we generally try to only include pretty well-established procedures, particularly those which are known to be effective.

If you have code to do this already, we would be thrilled to add it as a demo to Chemprop, if you'd like to contribute a PR.

**Comment by lewis-apheris** [comment]
@JacksonBurns thanks to you and @craabreu for looking into this - interesting work
